controllers/mongodb_users.go

```diff
@@ -0,0 +1,51 @@
+package controllers
+
+import (
+	"fmt"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/api/v1"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/secret"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r ReplicaSetReconciler) ensureUserResources(mdb mdbv1.MongoDBCommunity) error {
+	for _, user := range mdb.GetScramUsers() {
+		secretNamespacedName := types.NamespacedName{Name: user.PasswordSecretName, Namespace: mdb.NamespacedName().Namespace}
+		if _, err := secret.ReadKey(r.client, user.PasswordSecretKey, secretNamespacedName); err != nil {
+			if apiErrors.IsNotFound(err) {
+				r.log.Errorf(`User password secret "%s" not found`, secretNamespacedName)
+			}
+			return err
+		}
+
+		r.secretWatcher.Watch(secretNamespacedName, mdb.NamespacedName())
+	}
+
+	return nil
+}
+
+func (r ReplicaSetReconciler) updateConnectionStringSecrets(mdb mdbv1.MongoDBCommunity) error {
+	for _, user := range mdb.GetScramUsers() {
+		secretNamespacedName := types.NamespacedName{Name: user.PasswordSecretName, Namespace: mdb.NamespacedName().Namespace}
+		pwd, err := secret.ReadKey(r.client, user.PasswordSecretKey, secretNamespacedName)
+		if err != nil {
+			return err
+		}
+
+		operatorSecret := secret.Builder().
+			SetName(fmt.Sprintf("mdbc-%s-%s", user.Database, user.Username)).
```

Comment:
I almost think we could extract this `fmt.Sprintf` call to a method inside `user`. We also duplicate this in the testing. What do you think?

Reply:
implemented, good idea